gazebo fails to compile against graphviz 9.0

Homebrew-core recently merged version 9.0 of Graphviz, which is causing gazebo-classic build failures.

## Environment
* OS Version: macOS: 11.7.7-x86_64
* Source or binary build? source build of gazebo11, af7d3278a1cfae54328bfd2d4eba7b1c29ee669c

## Description
* Expected behavior: gazebo-classic should compile against graphviz 9.0
* Actual behavior: build error

## Steps to reproduce
<!-- Provide steps so we can try to reproduce this issue -->

1. Run https://build.osrfoundation.org/job/gazebo-ci-gazebo11-homebrew-amd64

OR

1. Install graphviz 9.0
2. Build gazebo-classic
3. Observe error

## Output

[![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=gazebo-ci-gazebo11-homebrew-amd64&build=186)](https://build.osrfoundation.org/job/gazebo-ci-gazebo11-homebrew-amd64/186/) https://build.osrfoundation.org/job/gazebo-ci-gazebo11-homebrew-amd64/186/

~~~
[ 77%] Building CXX object gazebo/gui/CMakeFiles/gazebo_gui.dir/qgv/private/QGVCore.cpp.o
In file included from /Users/jenkins/jenkins-agent/workspace/gazebo-ci-gazebo11-homebrew-amd64/gazebo-classic/gazebo/gui/qgv/private/QGVCore.cpp:19:
/Users/jenkins/jenkins-agent/workspace/gazebo-ci-gazebo11-homebrew-amd64/gazebo-classic/gazebo/gui/qgv/private/QGVCore.h:105:12: error: no member named 'mem' in 'Agdisc_s'
      disc.mem = &AgMemDisc;
      ~~~~ ^
/Users/jenkins/jenkins-agent/workspace/gazebo-ci-gazebo11-homebrew-amd64/gazebo-classic/gazebo/gui/qgv/private/QGVCore.h:105:19: error: use of undeclared identifier 'AgMemDisc'
      disc.mem = &AgMemDisc;
                  ^
2 errors generated.
~~~

Related API break in https://gitlab.com/graphviz/graphviz/-/commit/0f6504e90775c7259b192d382bfe910e4e98dcef

--- a/cmake/SearchForStuff.cmake
+++ b/cmake/SearchForStuff.cmake
@@ -771,6 +771,9 @@ if (NOT GRAPHVIZ_FOUND)
 else ()
   message (STATUS "Looking for libgraphviz-dev - found")
   set (HAVE_GRAPHVIZ ON CACHE BOOL "HAVE GRAPHVIZ" FORCE)
+  if (${GRAPHVIZ_CGRAPH_PKG_VERSION} VERSION_LESS 9.0)
+    set(GRAPHVIZ_VERSION_LT_9 TRUE)
+  endif ()
 endif ()
 
 ########################################

--- a/gazebo/gui/CMakeLists.txt
+++ b/gazebo/gui/CMakeLists.txt
@@ -206,6 +206,11 @@ add_dependencies(gzclient gazebo_gui)
 target_compile_definitions(gazebo_gui
   PRIVATE BUILDING_DLL_GZ_GUI
 )
+if (GRAPHVIZ_VERSION_LT_9)
+  target_compile_definitions(gazebo_gui
+    PRIVATE GRAPHVIZ_VERSION_LT_9
+  )
+endif()
 
 target_link_libraries(gazebo_gui
   libgazebo_client

--- a/gazebo/gui/qgv/private/QGVCore.h
+++ b/gazebo/gui/qgv/private/QGVCore.h
@@ -102,7 +102,10 @@ class QGVCore
       rdr.len = strlen(cp);
       rdr.cur = 0;
 
+#ifdef GRAPHVIZ_VERSION_LT_9
+      // This variable only exists for versions before Graphviz 9.0
       disc.mem = &AgMemDisc;
+#endif
       disc.id = &AgIdDisc;
       disc.io = &memIoDisc;
       g = agread(&rdr, &disc);

